packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -346,14 +348,58 @@ Future<XcodeBuildResult> buildXcodeProject({
     );
   }
 
-  final Status buildStatus =
-      logger.startProgress('Running Xcode build...', expectSlowOperation: true);
+  Status buildSubStatus;
+  Status initialBuildStatus;
+  Directory scriptOutputPipeTempDirectory;
+
+  if (logger.supportsColor) {
+    scriptOutputPipeTempDirectory = fs.systemTempDirectory
+        .createTempSync('flutter_build_log_pipe');
+    final File scriptOutputPipeFile =
+        scriptOutputPipeTempDirectory.childFile('pipe_to_stdout');
+    os.makePipe(scriptOutputPipeFile.path);
+
+    Future<void> listenToScriptOutputLine() async {
+      final List<String> lines = await scriptOutputPipeFile.readAsLines();
```

Comment:
Would it be better to use `openRead()` and stream the lines as they come?

Reply:
Discussed offline. 
It doesn't seem to work
https://github.com/dart-lang/sdk/issues/32191

For now, we'll never print more than one from the other end of the pipe so it should be ok.